octopus-server/src/test/java/octopus/teamcity/server/generic/CreateDeploymentStepTest.java

```diff
@@ -0,0 +1,143 @@
+package octopus.teamcity.server.generic;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import jetbrains.buildServer.serverSide.InvalidProperty;
+import octopus.teamcity.common.createdeployment.CreateDeploymentPropertyNames;
+import org.junit.jupiter.api.Test;
+
+class CreateDeploymentStepTest {
+
+  private final CreateDeploymentStep step = new CreateDeploymentStep();
+
+  @Test
+  public void describePropertiesProducesExpectedOutput() {
+    String message = step.describeParameters(buildPropertiesMap());
+    assertEquals(
+        "Project: ProjectName\nEnvironments: Env1, Env2\nRelease version: 1.0.0-SNAPSHOT", message);
+  }
+
+  @Test
+  public void validatePropertiesReturnsIllegalArgumentExceptionOnNullPropertyValue() {
```

Comment:
these tests shouldn't throw, rather it needs to return a list of "InvalidProperties"

Reply:
Deployment step refactored and associated tests updated